wagtail/admin/viewsets/chooser.py

```diff
@@ -34,12 +34,34 @@ class ChooserViewSet(ViewSet):
     #: The view class used after an item has been chosen; must be a subclass of ``wagtail.admin.views.generic.chooser.ChosenView``.
     chosen_view_class = chooser_views.ChosenView
 
+    #: The view class used to handle submissions of the 'create' form; must be a subclass of ``wagtail.admin.views.generic.chooser.CreateView``.
+    create_view_class = chooser_views.CreateView
+
     #: The base Widget class that the chooser widget will be derived from.
     base_widget_class = BaseChooser
 
     #: Defaults to True; if False, the chooser widget will not automatically be registered for use in admin forms.
     register_widget = True
 
+    #: Form class to use for the form in the "Create" tab of the modal.
+    creation_form_class = None
+
+    #: List of model fields that should be included in the creation form, if creation_form_class is not specified.
+    form_fields = None
+
+    #: List of model fields that should be excluded from the creation form, if creation_form_class.
+    #: If none of ``creation_form_class``, ``form_fields`` or ``exclude_form_fields`` are specified, the "Create" tab will be omitted.
+    exclude_form_fields = None
+
+    search_tab_label = _("Search")  #: Label for the 'search' tab in the chooser modal
+    create_action_label = _(
+        "Create"
+    )  #: Label for the submit button on the 'create' form
+    create_action_clicked_label = None  #: Alternative text to display on the submit button after it has been clicked
+    creation_tab_label = None  #: Label for the 'create' tab in the chooser modal (defaults to the same as create_action_label)
+
+    permission_policy = None
```

Comment:
question: all other properties are document. this one is not. is that because it is a somewhat internal detail?

Reply:
Yep, pretty much. Obviously to document this we'll have to explain what a permission policy _is_, and I'm reluctant to go down the path of making that a fully official public-facing API. We're going to have to find some middle ground there though, at least as far as giving people an official way to enable collection-based permissions - otherwise we fall short of the goal of letting people create a wagtailmedia equivalent using only public APIs.

Reply:
🐰🕳️